tests/common/tf/retrieval/retrieval_utils.py

```diff
@@ -0,0 +1,575 @@
+#
+# Copyright (c) 2021, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import logging
+import os
+import time
+from dataclasses import dataclass
+from typing import Any, Optional
+
+import numpy as np
+import tensorflow as tf
+import wandb
+from tensorflow.keras import regularizers
+from tensorflow.keras.utils import set_random_seed
+
+import merlin.models.tf as mm
+from merlin.io.dataset import Dataset
+from merlin.models.tf.blocks.core.transformations import PopularityLogitsCorrection
+from merlin.models.utils import schema_utils
+from merlin.schema.tags import Tags
+
+
+def filter_schema(schema, filter_dataset_features=""):
+    schema = schema.select_by_tag([Tags.ITEM_ID, Tags.USER_ID, Tags.ITEM, Tags.USER])
+
+    if filter_dataset_features:
+        schema = schema.select_by_name(filter_dataset_features.split(","))
```

Comment:
Now that we use Fiddle instead of a CLI, would it make sense to change this to a list instead of a string that needs to be splitted?

Reply:
Hey @marcromeyn . I think we can keep it as a string so that if we decide later to create a CLI based on Fiddle to run these unit tests with different arguments the string arg would be easier to parse that a list. what do you think?